iroh-api/src/api.rs

```diff
@@ -66,17 +89,17 @@ impl Api {
     }
 
     /// High level get, equivalent of CLI `iroh get`.
-    pub fn get(
-        &self,
+    pub fn get<'a>(
+        &'a self,
         ipfs_path: &IpfsPath,
-    ) -> Result<LocalBoxStream<'static, Result<(RelativePathBuf, OutType)>>> {
+    ) -> Result<LocalBoxStream<'a, Result<(RelativePathBuf, OutType)>>> {
```

Comment:
on rebase I needed to switch the lifetime param of the return value from `'static` to a new generic. I'm not totally clear on why this is, and think it's a bug. Given that we clone the resolver below I don't see why the lifetime of the return value should need to match that of `self`.

Reply:
the problem is on (new) line 122 & 128: you're cloning resolver from `self` again, which pulls `self` into the stream future.  But you already clone `resolver` out of `self` on (new) line 102 so you don't need to modify those two lines at all: using `resolver.clone()` instead of `self.resolver.clone()` there solves the lifetime issue.